libs/npm/core/src/components/table/useTable.tsx

```diff
@@ -0,0 +1,190 @@
+import React from 'react';
+
+import {
+  ITableHook,
+  ITableInternal,
+  ITableInternalRow,
+  ITablePage,
+  ITableSort,
+  TableInternal,
+} from '.';
+
+/**
+ * This hook generates a table object and manages state mutations.
+ * Only the following properties will mutate externally; columns, data, options, paging, sorting.
+ * @param param0 Hook properties
+ * @returns A TableInternal object instance.
+ */
+export const useTable = <T extends object>({
```

Comment:
I am not a fan of how I implemented state management with this hook.  It seems to work okay for now though.